subprojects/process-services/src/main/java/org/gradle/process/internal/health/memory/JvmMemoryInfo.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal.health.memory;
+
+public interface JvmMemoryInfo extends JvmMemoryStatus {
```

Comment:
This is an odd relationship, same for `OsMemoryInfo`. I would remove this `extends` clause so that `JvmMemoryInfo` produces `JvmMemoryStatus` snapshots. I would also change the contract of `JvmMemoryStatus` so that it is immutable and always represents a snapshot.

Same kind of thing for `OsMemoryInfo`.